recipe/meta.yaml

```diff
@@ -28,7 +28,6 @@ requirements:
     - nlohmann_json     # [build_platform != target_platform]
     - cppzmq            # [build_platform != target_platform]
     - zlib              # [build_platform != target_platform]
-    - zstd-static =1.5.5
```

Comment:
It seems we removed this dependency here. @certik I am curious to know why are we needing it in [Install zstd-static for osx/arm](https://github.com/conda-forge/lfortran-feedstock/pull/59/commits/f25ebfb8bd42a3919474b405c8612afa55162d34)?

Reply:
Because of this: https://github.com/conda-forge/lfortran-feedstock/pull/59#issuecomment-1759193749

Reply:
>Because of this: https://github.com/conda-forge/lfortran-feedstock/pull/59#issuecomment-1759193749

Yes, I understand. I was able to reproduce it here https://github.com/conda-forge/lfortran-feedstock/pull/55#issuecomment-1759379054. I am curious to know why it is being needed with LLVM/Clang 16.

Reply:
They started depending on zstd from LLVM 16 I believe.